overlord/snapstate/snapmgr.go

```diff
@@ -510,6 +521,8 @@ func (m *SnapManager) ensureRefreshes() error {
 	}
 	chg.Set("snap-names", updated)
 	chg.Set("api-data", map[string]interface{}{"snap-names": updated})
+
+	m.nextRefresh = nil
```

Comment:
AIUI a go timer is only called once (c.f. https://play.golang.org/p/OuAh1Ua6db). A ticker would be callled more often. So I think we don't need to cancel.